dev/release/05-binary-upload.sh

```diff
@@ -104,6 +105,28 @@ fi
 if [ ${UPLOAD_PYTHON} -gt 0 ]; then
   rake_tasks+=(python:rc)
 fi
+if [ ${UPLOAD_R} -gt 0 ]; then
+  rake_tasks+=(r:rc)
+  # The R libarrow binaries have to be converted to the correct dir structure
+  pushd $artifact_dir/r-binary-packages
+  # R package binaries are distributed via CRAN
+  rm  r-pkg*
+  r_src_files=$(ls r-lib*)
+  # Decode relative path from file name.
+  # r-lib__libarrow__bin__centos-7__arrow-8.0.0.zip 
+  # --> libarrow/bin/centos-7/arrow-8.0.0.zip
+  r_dest_files=($(echo $r_src_files | sed 's/r-lib__//g; s/__/\//g'))
+  r_src_files=($r_src_files)
+
+  mkdir -p $(dirname ${r_dest_files[@]})
+  r_length=${#r_dest_files[@]}
+
+  for (( i=0; i<${r_length}; i++ ));
+  do
+    mv ${r_src_files[$i]} ${r_dest_files[$i]}
+  done
+  popd
```

Comment:
I want to move this logic to `binary-task.rb`. If we do this at here, we can't run this script multiple times.

I'll provide a diff for it later.

Reply:
Ok, I could also use `cp` and add a clean up step in the beginning? That way it would still be repeatable, but feel free to add a diff for the ruby version (or add it in a follow up?)

Reply:
I have added this for now.